lib/facility.rb

```diff
@@ -1,14 +1,63 @@
 class Facility
-  attr_reader :name, :address, :phone, :services
-
-  def initialize(name, address, phone)
-    @name = name
-    @address = address
-    @phone = phone
+  attr_reader :name, :address, :phone, :services, :registered_vehicles, :collected_fees
+  def initialize(info)
+    @name = info[:name]
+    @address = info[:address]
+    @phone = info[:phone]
     @services = []
+    @registered_vehicles =[]
+    @collected_fees = 0 
   end
 
-  def add_services(service)
+  def add_service(service)
+    #require 'pry';binding.pry
     @services << service
   end
-end
+
+  def register_vehicle(vehicle)
+    if services.include?("Vehicle Registration")
+      vehicle.registration_date = Date.today
+      @registered_vehicles << vehicle
+
+      if vehicle.antique? 
+        vehicle.plate_type = :antique
+        @collected_fees += 25
+      elsif vehicle.electric_vehicle? == true 
+        vehicle.plate_type = :ev
+        @collected_fees += 200
+      else   
+        vehicle.plate_type = :regular
+        @collected_fees += 100  
+      end 
+      registered_vehicles
+    end 
+  end 
+
+  def administer_written_test(registrant)
+    if services.include?("Written Test") && registrant.permit? == true && registrant.age >= 16
+      registrant.license_data[:written] = true
+    else 
+      return false 
+
+    end 
+  end 
+  def administer_road_test(registrant)
+    #require 'pry';binding.pry
+    if services.include?("Road Test") && registrant.permit? && registrant.license_data[:written] == true
+      registrant.license_data[:license] = true
+ 
+    end 
+    registrant.license_data[:license]
+  end
+
+  def renew_drivers_license(registrant)
+    #require 'pry';binding.pry
+    if services.include?("Renew License") && registrant.permit? && registrant.license_data[:written] == true && registrant.license_data[:license] == true
+      registrant.license_data[:renewed] = true
+ 
+    end 
+    registrant.license_data[:renewed]
+  end
+end    
```

Comment:
Overall this is a well-put-together class. Just a few suggestions:

There are several places within this class where you are doing `== true`. This is not required and you can simplify this condition by removing this unnecessary check, (ie. `&& registrant.permit? &&`, `registrant.license_data[:written]`)

There also seems to be some inconsistency in coding style throughout this class. On line 40 you explicity return `false` but on lines 50 and 59 you don't and instead return the value in the `license_data` hash. The `administer_written_test` method is also the only one that uses an `else` where the others do not.

My point is, no matter how small or large the codebase, the code writing should be consistent and it should look like 1 person wrote it, even in group work.